### ISSUE TITLE
Support .NET Standard

### ISSUE THREAD

**Issue by reisenberger** [dennisdoomen/FluentAssertions#431]
It would be useful to have a FluentAssertions nuget package supporting .NET Standard targets (ideally 1.0, given the level of PCL support FluentAssertions currently brings).  

For libraries such as [Polly](https://github.com/App-vNext/Polly) whose tests use FluentAssertions, we are at present blocked from releasing .NET Standard libraries _unit-tested within the same target framework they are built for_.

If you want anyone to help test pre-release versions supporting .NET Standard in due course, very happy to do that.  

(Appreciate this is probably an on-going process for FluentAssertions which could be affected by the speed of .NET Standard support by underlying XUnit, NUnit etc?)

Thanks for the _fabulous_ library!

**Comment by dennisdoomen** [comment]
Hi @reisenberger. Thanks for the appreciation. I don't quite understand your problem. If I'm not mistaken (and @onovotny can confirm this), PCL or .NET Standard compliant assemblies are just compiled against a (partial) shim. So if you want to make sure your library works perfectly for a particular target, you must create unit test projects for every supported target. Is that what you're doing? If so, for which target does FA not work right now?

**Comment by onovotny** [comment]
I have to agree that I don't quite understand the issue either. ".NET Standard" is like an *uber* PCL. There are a bunch of runtimes, like .NET Core, UWP, .NET Framework, Xamarin, that support all of it.

If what you want to do is test your code on each platform, then you can do that easily by creating a unit test project on each and then referencing your code from there.

**Comment by reisenberger** [comment]
Hi @dennisdoomen and @onovotny, thanks very much both for the quick responses.  I may not have expressed myself clearly (and might have some wrong assumptions); I'll try to be more specific.  

Prior to the existence of the .NET Standard tfms, Polly and its tests (using FluentAssertions and XUnit) were (and still are) constructed with unit test projects for specific supported targets.  So for `Polly.Net45.csproj` we had (and still have) `Polly.Net45.specs.csproj`; for `Polly.Pcl` we have `Polly.Pcl.Specs` [* maybe the source of confusion on my part - see below]

What I was seeking to do was convert both the projects `Polly.Pcl` and `Polly.Pcl.Specs` from targeting PCL 259 to target .NET Standard 1.0, following your [great](https://oren.codes/2016/06/23/portable-is-dead-long-live-netstandard/) [articles](https://oren.codes/2016/02/08/project-json-all-the-things/) @onovotny . When I do this `Polly.Pcl` was an easy convert to .NET Standard, but with `Polly.Pcl.Specs` I hit the block of no nuget packages for XUnit runners and Fluent Assertions targeting .NET Standard 1.0.

My assumption (this may have been the incorrect bit) was that to unit-test a library to be offered on nuget as supporting .NET Standard 1.0, I ought to have a specs project also targeting .NET Standard 1.0 (just as we had Polly.Pcl and Polly.Pcl.specs [*] in the Polly project layout we inherited from Michael Wolfenden).  I think I am hearing from your preceding explanations that we should have test projects targeting each runtime instead, ie `Polly.Net45.specs` (as we do) etc.  Thus, for unit-testing support for .NET Core 1.0 RTM we should be constructing a `Polly.NetCore1.0.specs` rather than `Polly.NetStandard1.6.specs` and so on (even if the support for .NET Core 1.0 RTM is released to nuget as a .NETStandard1.6-compliant package).  Is that it?

This makes sense to me.  I get that both PCL and the .NET Standards are (if I have it right) a specification of a set of APIs that a given platform/runtime might fulfil.  I may have been misled by the Polly project structure we inherited having a `Polly.Pcl.Specs` project (PCL 259), and thinking this therefore ought to be converted to `Polly.NetStandard10.specs` targeting Net Standard 1.0.

V grateful for clarification if this is more along the right lines
